src/AutoRest.CSharp/Common/Input/Source/SourceInputModel.cs

```diff
@@ -45,13 +45,13 @@ public ModelTypeMapping CreateForModel(INamedTypeSymbol? symbol)
             return new ModelTypeMapping(_modelAttribute, _schemaMemberNameAttribute, symbol);
         }
 
-        public INamedTypeSymbol? FindForType(string ns, string name)
+        public INamedTypeSymbol? FindForType(string ns, string name, bool includeArmCore = false)
         {
             var fullyQualifiedMetadataName = $"{ns}.{name}";
             if (!_nameMap.TryGetValue(name, out var type) &&
                 !_nameMap.TryGetValue(fullyQualifiedMetadataName, out type))
             {
-                type = _compilation.Assembly.GetTypeByMetadataName(fullyQualifiedMetadataName);
+                type = includeArmCore ? _compilation.GetTypeByMetadataName(fullyQualifiedMetadataName) : _compilation.Assembly.GetTypeByMetadataName(fullyQualifiedMetadataName);
             }
```

Comment:
@m-nash I saw you fixed this bug so FindForType() does not look for classes in Azure.ResourceManager package, but as we don't have the code for removing orphaned models yet, I need to use the previous code for finding duplicate models in Resources (they are orphaned as their parent classes are replaced by PropertyReferenceType), so I added includeArmCore parameter and use it in https://github.com/Azure/autorest.csharp/blob/0aacf2bffb00ab9c8f5705fa4002381b1610f511/src/AutoRest.CSharp/Mgmt/AutoRest/MgmtTarget.cs#L29-L35

This is a temporary fix but as I'm not familiar with compilation and assembly in C#, do you have suggestions for better ways to do this?